### DISCUSSION TITLE
A better way for Chord Button ? Let's See

### DISCUSSION THREAD

**Discussion by javegon** [bangcorrupt/hacktribe#211] [Ideas]
***UNDER REVISION***
_I am still figuring out what do the machine in each Chord set and Scale._
_But It doesn't allow you to change the scale or key once the chords were recorded in the sequence when they were created with Chord button , only when we recorded chords using Keyboard ( manually ) will be affected by changes in Key and Scale _

That is the chart how the Chord Set works

>1. Third (1, 3)
>2. Triad (1, 3, 5)
>3. Major Seventh (1, 3, 5, 7)
>4. Ninth (1, 3, 5, 9)
>5. Dominant Seventh (1, 5, 7b, 3)  <-- With an open voicing on the 3rd

I guess would have been really great to have the right progress of chord for each key note. I mean... With his minor, major and disminished chords for each key note using through the menu. Key note C, scale minor, chord(a kind of harmony) The progression of chord using the pads ( not all 7th or all 9th ) Because have all 7th doesn't have much sense. 

I try to explain in a graphic mode

A) We select key note C, Scale Aeolian, Chord Set 2 (Triad) Then we get :

> Cm Dm Em Fm Gm Am Bm

B)That is not useful to make a song at all, because a real progression of Cm is : 

> Cm Ddim Eb Fm Gm Ab Bb

Now we chan see how limited the Chord Set is with Scale Selected in order to make a melody.
_*Actually in minor(Aeolian) and major scales works good, but it is an example, in order to make it easer to explain it._ Here is how it really works per Pad num.

> Cm Ddim Eb Fm Gm Ab Bb Cm

Well the idea I have is divided in two options

1. it would be great to create more option in Chord Set, with more Chord Sets like Maj, Minor, diminished, sus4, 7th, etc... it will be an option to have all the possibilities. Maybe all chord would be too much but add diminished, augmented,  major, major7, minor, minor7 and could be great.
2. **It is how it works actually** (Other way would be create a real progression like the example with Cm in B) paragraph. Setting the Chord button with the real progression in combination with the Key note and Scale. playing the pad num with the correct chord in each scale and key note). Obviously this would be very limited in order to have nice harmonies or more sophisticated progression like jazzy progression, but then the can use the chord set in other way than it is using right now. Maybe setting them like 1.RegularProgression(depending of the key note and Scale) 2.Third 3.Triad 4. 7th 5.9th 6.11th 7.Dim 8.Aug 9.Sus4 10.Dom7th 11. 1rst Inversion ... and so on.

Well... maybe someone else ,with more knowledge about music than me, can add some ideas or corrections, or better way to create better using of this.

**Comment by BKLronin** [comment]
I think its easier to just play them on the pads. I mean if its in scale mode you can just play each chord by counting the pads. Ive never used chordmode much on any device because of what yo said. 

I mean on the cycles you can simply set chord and inversion per step but a whole machine aka synth section needed to be used for this. Its not availabel across the whole device. As everything else is mono voice.

**Reply by javegon** [reply]
I found that in chord mode, the chord progression respect the scale it is works correctly for minor and major. but it is not affected once was recorded with Chord mode by the change of scale or key.

**Reply by BKLronin** [reply]
Hm, havent tested in in depth. I thin k I made one track just by ear with chord mode a long time ago. Is ounded correct to me.

I think the easiest solution would be to have the pads as chromatic keyboards with black key lighted up red and white keys blue or something. That way you dont need to connect ka keyboard or fiddle with chord mode :)

**Reply by javegon** [reply]
I found that works fine. I tested minor (Aeolian), major (Ionian) and harmonic minor scale (phrygian). Then I could say every Scale and Key works fine with the Chord button. Maybe add same sus4 , aug or other type or chord in Chord Set . 
So...nothing more to add.

**Reply by BKLronin** [reply]
It automatically uses major and diminished chords when necessary? :O

Inversions are sadly missing though :( Thats why I would prefer just the chromatic keyboard mapping so there is no restriciton. Many pad devices have this these days.

**Reply by javegon** [reply]
yes, major and diminished works when it necessary in all scales. That is really great. 

Inversions and 7th or 9th and other can be modified in step edit, or using the Chord set. 

A keyboard is the best way if I would know how to play a it :P Chord is it a great solution.

**Reply by BKLronin** [reply]
`Inversions and 7th or 9th and other can be modified in step edit`
Ah yes, forgot that as the chord is not baked in to the step but actually played and recorded with notes.  

Just made a nice electro summerhit with minor chords and arpeggio :D

True scale playback is way faster and a very good way if you havent learned normal keyboard use.

**Comment by bangcorrupt** [comment]
> major and diminished works when it necessary in all scales

Currently this only works for the original factory scales, the scales I added all use the chord set from the chromatic scale.  Some of the scales use diatonic chord sets, others (like the Raga) have drone chords.  

In the latest version I enabled all 8 chord sets from the factory firmware, I haven't checked which chords they are.  I think there is also a bug so they don't cycle through properly and end up out of order.

At some point I'd like to make scales and chords user configurable.
